### PR TITLE
feat: add mtp bootstrap embedding context.

### PR DESCRIPTION
**Pull request by phantomlei3**
## Description

This PR adds the first part of MTP bootstrap support for disaggregated PD handoff.

In PD mode, the decode side needs the target hidden state for the handoff token before the first MTP decode step can follow the normal draft-extend path. This change introduces bootstrap embedding storage and transport plumbing so that the handoff embedding can be preserved, passed through `ForwardInput`, and written into the MTP embedding cache before decode validation.

Main changes:
  - Store temporary MTP bootstrap embeddings on `Sequence` and capture them from sample outputs.
  - Add `EmbeddingCache::write_mtp_bootstrap_context` to materialize a valid decode state for the first MTP decode step.
  - Extend `ModelEmbeddingInput`, contiguous input buffering, and shared-memory serialization with bootstrap row indices and embeddings.
  - Update `MTPWorkerImpl` to preserve selected prefill embeddings for handoff and seed the embedding cache from bootstrap inputs during decode.
  - Add regression coverage for `Sequence`, `Batch`, and `EmbeddingCache` bootstrap storage behavior.

## Change Type

- [ ] Bug fix
- [x] New feature
- [ ] Performance improvement
- [x] Refactor
- [ ] Documentation
- [ ] Test
- [ ] Build or CI

## Pull Request Checklist

Thank you for contributing to xLLM. Before requesting review, please make sure the following items are complete.

### PR Title and Commit Messages

- [x] The PR title and each commit message follow the xLLM commit format: `<type>: <subject>`.

> Allowed types: feat, bugfix, docs, test, refactor, chore, style, revert, perf, model, build, release.
> The subject should use clear English, start with a verb, include at least 4 words, and end with `.`.

### Pre-commit Checks

- [x] I have installed `pre-commit` by running `pip install pre-commit` or an equivalent command.
- [x] I have installed the hooks with `pre-commit install`.
- [x] I have run `pre-commit run --all-files` and fixed any reported issues.

> If you are unsure how to set up `pre-commit`, see [the pre-commit documentation](https://pre-commit.com/).

### Self Review

- [x] I have self-reviewed the code according to `.agents/skills/code-review/references/custom-code-style.md`, especially code written or assisted by AI.
- [x] I have rebased this PR onto the latest `main` branch.

### Build and Test Coverage

- [x] Tests have been added or updated as needed.
- [ ] CUDA: `python setup.py build test` has passed on a CUDA machine.
- [ ] NPU: `python setup.py build test` has passed on an NPU machine.
- [x] MLU: `python setup.py build test` has passed on an MLU machine.

### CODE REVIEW

File: xllm/core/framework/kv_cache/embedding_cache.cpp [hunk 1]
```diff
@@ -96,6 +96,26 @@ void EmbeddingCache::write_prefill_target_context(
...
+  CHECK(embedding.defined()) << "MTP bootstrap embedding is undefined";
+  CHECK_GE(token_id, 0) << "MTP bootstrap token should be valid";
+
+  DecodeState state;
+  state.valid = true;
+  state.request_id = request_id;
+  state.all_draft_accepted = false;
+  state.token_id = token_id;
+  state.position_offset = 0;
+  state.embedding = embedding.detach().clone();
```
[the remark this answers is not in the document]
Comment (phantomlei3) [reply xllm/core/framework/kv_cache/embedding_cache.cpp]: fixed

File: xllm/core/runtime/mtp_worker_impl.cpp [hunk 2]
```diff
@@ -715,14 +752,27 @@ std::optional<ForwardOutput> MTPWorkerImpl::step_prefill(
...
+    if (bootstrap_embeddings.size(0) !=
+        static_cast<int64_t>(
+            input.input_params.embedding.embedding_ids.size())) {
+      torch::Tensor bootstrap_idxes =
+          input.sampling_params.selected_token_idxes.to(
+              torch::dtype(torch::kLong).device(bootstrap_embeddings.device()));
+      bootstrap_embeddings =
+          bootstrap_embeddings.index_select(/*dim=*/0, bootstrap_idxes);
+    }
+    output.sample_output.embeddings = bootstrap_embeddings.detach().clone();
```
[the remark this answers is not in the document]
Comment (phantomlei3) [reply xllm/core/runtime/mtp_worker_impl.cpp]: fixed

File: tests/core/framework/kv_cache/embedding_cache_test.cpp [hunk 3]
```diff
@@ -142,4 +142,32 @@ TEST(EmbeddingCacheTest, RequestMismatchMaterializesMissingState) {
...
+  EXPECT_EQ(states[0].request_id, "req_bootstrap");
+  EXPECT_EQ(states[0].token_id, 17);
+  EXPECT_EQ(states[0].position_offset, 0);
+  EXPECT_FALSE(states[0].all_draft_accepted);
+  EXPECT_TRUE(tensor_equal(states[0].embedding, embedding));
+
+  embedding.fill_(9.0f);
+  states = cache.read_decode_states({1}, {"req_bootstrap"});
+  EXPECT_TRUE(
+      tensor_equal(states[0].embedding, torch::tensor({1.0f, 2.0f, 3.0f})));
```
[the remark this answers is not in the document]
Comment (phantomlei3) [reply tests/core/framework/kv_cache/embedding_cache_test.cpp]: fixed

**Review by RobbieLeung — APPROVED** [review]

**Review by XuZhang99 — APPROVED** [review]
